### PR TITLE
Eclipse instructions

### PR DESCRIPTION
**Pull request by heplesser**
I have revised and significantly extended the NEST & Eclipse instructions. I hope someone could give it a try on Linux as well as OS X. I also hope that the threshold for merging documentation of this kind is not too high, as I believe the document is useful as it is, and I need to attend to other matters now rather than perfecting this doc.

### CODE REVIEW

File: eclipse_workflow.md [hunk 1]
```diff
@@ -0,0 +1,305 @@
...
+1. select `Disable auto build` (because that only builds, but does not install)
+
+You can now run NEST by clicking the "Play" button. Input is echoed in
+a slightly funny way in the build-in console, but NEST works fine. You
+need to quit NEST with the `quit` command, `Ctrl-D` does not seem to
+work (made my machine hang totally on one occasion).
+
+#### Running PyNEST from Eclipse
+
+To be written.
```
Comment (simonscompneurostuff) [inline eclipse_workflow.md]: very interested in seeing instructions for this. I'd love to be able to set up and start a simulation from python, and then debug it graphically (expanding objects in a tree view, viewing call stack, etc.) using eclipse.

**Comment by abigailm** [comment]
I don't use eclipse, but the document itself reads very cleanly. +1 from me, but it would be great if an eclipse user could test it.

File: eclipse_workflow.md [hunk 2]
```diff
@@ -0,0 +1,305 @@
...
+
+### Debugger under OSX
+
+Under OSX, you need to install `gdb` to be able to debug NEST with
+Eclipse, since Eclipse does not support `lldb` (yet). For installation
+instructions, see
+
+ - [http://ntraft.com/installing-gdb-on-os-x-mavericks](http://ntraft.com/installing-gdb-on-os-x-mavericks)
+ - [https://sourceware.org/gdb/wiki/BuildingOnDarwin](https://sourceware.org/gdb/wiki/BuildingOnDarwin)
+
```
Comment (tammoippen) [inline eclipse_workflow.md]: Further you can use [Homebrew](http://brew.sh/)

    brew install homebrew/dupes/gdb

File: eclipse_workflow.md [hunk 3]
```diff
@@ -0,0 +1,305 @@
...
+1. check off for `Add to all languages`
+1. click `File system ...` and select the
+`$NEST_ROOT/bld_master_nompi/libnestutil` directory
+1. add the `$NEST_ROOT/bld_master_nompi/nest` directory in the same way
+1. rebuild the index when Eclipse suggest it or by choosing `Index >
+   Rebuild` from the context menu on the project.
+
+To enable code formatting with `clang-format` via `CppStyle`, open the Properties window
+for the project and go to `C/C++ General > Formatter`, enable project specific settings, choose `CppStyle` as Code Formatter. `Source > Format` will now format source code according to the `.clang-format` file shipped with NEST.
+
```
Comment (tammoippen) [inline eclipse_workflow.md]: CppStyle complains that it does not find a proper `.clang-format` file. You have to copy it into the workspace:

    cp $NEST_ROOT/src/.clang-format $HOME/eclipse/workspace/

**Pull request merged**